fvtest/compilertriltest/ArithmeticTest.cpp

```diff
@@ -167,3 +167,177 @@ INSTANTIATE_TEST_CASE_P(DivArithmeticTest, Int32Arithmetic, ::testing::Combine(
+
+TEST_P(Int8ToInt64, UsingLoadParam) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char *inputTrees = "(method return=Int64 args=[Int8] (block (lreturn (b2l (bload parm=0)))))";
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int64_t (*)(int8_t)>();
+    volatile auto exp = param.oracle(param.value);
+    volatile auto act = entry_point(param.value);
+    ASSERT_EQ(exp, act);
+}
+
+INSTANTIATE_TEST_CASE_P(ArithmeticTest, Int8ToInt64, ::testing::Combine(
+    ::testing::ValuesIn(TRTest::const_values<int8_t>()),
+    ::testing::Values(
+        std::make_tuple("b2l", b2l) )));
+
+class Int16ToInt32 : public TRTest::UnaryOpWithRetTypeTest<int32_t,int16_t> {};
+
+TEST_P(Int16ToInt32, UsingLoadParam) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char *inputTrees = "(method return=Int32 args=[Int16] (block (ireturn (s2i (sload parm=0)))))";
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int32_t (*)(int16_t)>();
+    volatile auto exp = param.oracle(param.value);
+    volatile auto act = entry_point(param.value);
+    ASSERT_EQ(exp, act);
+}
+
+INSTANTIATE_TEST_CASE_P(ArithmeticTest, Int16ToInt32, ::testing::Combine(
+    ::testing::ValuesIn(TRTest::const_values<int16_t>()),
+    ::testing::Values(
+        std::make_tuple("s2i", s2i) )));
+
+class Int16ToInt64 : public TRTest::UnaryOpWithRetTypeTest<int64_t,int16_t> {};
+
+TEST_P(Int16ToInt64, UsingLoadParam) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char *inputTrees = "(method return=Int64 args=[Int16] (block (lreturn (s2l (sload parm=0)))))";
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<int64_t (*)(int16_t)>();
+    volatile auto exp = param.oracle(param.value);
+    volatile auto act = entry_point(param.value);
+    ASSERT_EQ(exp, act);
+}
+
+INSTANTIATE_TEST_CASE_P(ArithmeticTest, Int16ToInt64, ::testing::Combine(
+    ::testing::ValuesIn(TRTest::const_values<int16_t>()),
+    ::testing::Values(
+        std::make_tuple("s2l", s2l) )));
+
+class Int32ToInt64 : public TRTest::UnaryOpWithRetTypeTest<int64_t,int32_t> {};
+
+TEST_P(Int32ToInt64, UsingLoadParam) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char *inputTrees = "(method return=Int64 args=[Int32] (block (lreturn (i2l (iload parm=0)))))";
```

Comment:
Could we format these in a "tree-like" form as you'd find in trace files? See [1] for example. It makes it much easier to read and it feels natural as if you were reading a trace log.

[1] https://github.com/eclipse/omr/blob/8a555b1f455096e71b9155989444ef9550da21a3/fvtest/compilertriltest/IfxcmpgeReductionTest.cpp#L97-L114